### PR TITLE
Emphasize pending pod effect while node grp not yet in backoff

### PR DESCRIPTION
**Pull request by vlerenc**
**How to categorize this PR?**
/area documentation

**What this PR does / why we need it**:
Describe in more detail what will happen to pending pods during a zone outage before the node group goes finally into backoff (cluster autoscaler).
Related, drop `nodeMonitorPeriod` that was mentioned in the touched sections and that was a mistake that went unnoticed originally.

**Which issue(s) this PR fixes**:
Confusion why Gardener/CA doesn't act and pending pods remain pending for very long during a zone outage (HA best practice guide).

**Special notes for your reviewer**:
@dguendisch @rishabh-11 Could you please take a look and if OK approve, so that we get the addition in quickly for the readers to be better informed when they pick the document up?

### CODE REVIEW

File: docs/usage/shoot_high_availability_best_practices.md
Comment (rfranzke) [inline docs/usage/shoot_high_availability_best_practices.md]:
```suggestion
It's critical to keep that in mind and accommodate for it. If you have already capacity up and running, the reaction time is usually much faster with leases (whatever you set) or endpoints (`spec.kubernetes.kubeControllerManager.nodeMonitorGracePeriod`), but if you depend on new/fresh capacity, the above should inform you how long you will have to wait for it and for how long pods might be pending (because capacity is generally missing and pending pods may have been "virtually scheduled" to new nodes that won't come up until the node group goes eventually into "back-off" and nodes in the healthy zones come up).
```

**Review by rfranzke — DISMISSED** [review]
Please also remove all other occurrences of `nodeMonitorPeriod` (this field does not exist in the `Shoot` API .

File: docs/usage/shoot_high_availability_best_practices.md
Comment (vlerenc) [reply docs/usage/shoot_high_availability_best_practices.md]: Well... nobody (including of course myself) noticed that in the original document. I fixed it now. Thanks.

**Review by dguendisch — COMMENTED** [review]
Nothing to add beyond what @rfranzke mentioned. The rest
/lgtm

**Comment by vlerenc** [comment]
Thank you for the reviews. Merged to get it into the docs asap.

**Comment by oliver-goetz** [comment]
@vlerenc please don't merge manually! This makes all tests of PRs which are approved and about to be merged starting from zero again before they can get merged. Thanks 😄

**Comment by vlerenc** [comment]
I am sorry @oliver-goetz. 😞

**Comment by oliver-goetz** [comment]
no worries 👍

**Pull request merged**
